### PR TITLE
chore(#8706): split integration test job

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

#8706

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8706-split-integration-test-job/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8706-split-integration-test-job/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8706-split-integration-test-job/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by garethbowen** [comment]
@dianabarsan Review please. I'm not excited about the names, eg: `ci-integration-all-all`. I think ideally integration tests would be in a separate matrix, and we should script most of what's in the yml now so we can call it from two different steps without duplication. But for now this achieves the goal of splitting the slowest job in half.

After merging I'll change the "required" builds to match the new names.

File: tests/integration/.mocharc-sentinel.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  asyncOnly: false,
+  spec: [
+    'tests/integration/sentinel/**/*.spec.js'
+  ],
+  timeout: 200 * 1000, //API takes a little long to start up
+  reporter: 'spec',
+  require: [ 'tests/integration/hooks.js' ],
+  captureFile: 'tests/results/results.txt',
+  exit: true,
+};
```
Comment (garethbowen) [inline tests/integration/.mocharc-sentinel.js:23]: Would be better if it inherited from a common mocharc file. I guess we could "import" the js and extend it? What do you think?

File: tests/integration/api/controllers/db-doc.spec.js [hunk 2]
```diff
@@ -786,7 +786,6 @@ describe('db-doc handler', () => {
               if (reportScenarios[idx].allowed) {
                 chai.expect(result).to.deep.include(reportScenarios[idx].doc);
               } else {
-                console.log(idx, result);
```
Comment (garethbowen) [inline tests/integration/api/controllers/db-doc.spec.js:789]: Request objects are huge and logging them here accounted for most of the lines in the log file...

File: tests/integration/.mocharc-sentinel.js (hunk 1 above)
Comment (dianabarsan) [reply tests/integration/.mocharc-sentinel.js:23]: I definitely would prefer to require and extend it.

**Review by dianabarsan — APPROVED** [review]
Looking great pending the improvement about extending existent mocharc

File: tests/integration/api/controllers/db-doc.spec.js (hunk 2 above)
Comment (dianabarsan) [reply tests/integration/api/controllers/db-doc.spec.js:789]: This was probably accidentally committed. thanks for the cleanup!!

**Pull request merged**